isthmus/src/main/java/io/substrait/isthmus/TypeConverter.java

```diff
@@ -71,7 +71,7 @@ private Type toSubstrait(RelDataType type, List<String> names) {
       case SMALLINT -> creator.I16;
       case INTEGER -> creator.I32;
       case BIGINT -> creator.I64;
-      case FLOAT -> creator.FP32;
+      case REAL -> creator.FP32;
       case DOUBLE -> creator.FP64;
```

Comment:
Somewhat cursedly, even though FLOAT is a synonym for DOUBLE, within Calcite they are still 2 different SqlTypeNames. If we create tables using `FLOAT` now, the type conversion fails because of the now missing conversion.

Could you add a mapping for FLOAT to `FP64` along with a test to confirm that we can handle mapping `SqlTypeName.FLOAT` to it. `CalciteTypeTest.java` would be a good file for the test.

Reply:
I see, but the problem comes to how do we map FP64 back to the calcite type, it can be FLOAT or DOUBLE. `testType` in CalciteTypeTest requires a 1 to 1 map.

Reply:
> how do we map FP64 back to the calcite type

That's a good question. Given that FLOAT is defined as the synonym for DOUBLE, and that we were already using DOUBLE as the returned type, I think keeping that as is makes the most sense.

> it can be FLOAT or DOUBLE. testType in CalciteTypeTest requires a 1 to 1 map.

I think having a test that checks that DOUBLE roundtrips to FP64 and back (which we already have) and then a separate check that FLOAT gets mapped to FP64 (which we don't have yet) is sufficient here.

Reply:
Oh, you already added this in
https://github.com/substrait-io/substrait-java/pull/261/files#diff-ed9b94983bf865c9d1fd5cb424c02e9b3ab3f04501a853d95a21ab28fbcdcd57R63-R70 ✨